src/hooks/useWalletCloudBackup.js

```diff
@@ -15,53 +16,93 @@ import logger from 'logger';
-            wallets[walletId]
-          );
+      let fetchedPassword = password;
+      if (latestBackup && !password) {
+        // We have a backup but don't have the password, try fetching password
+        fetchedPassword = await fetchBackupPassword();
+      }
 
-          backupFile = await backupWalletToCloud(password, wallets[walletId]);
-        } else {
-          logger.log(
-            'walletCloudBackup:: adding to icloud backup',
-            wallets[walletId],
-            latestBackup
-          );
-          backupFile = await addWalletToCloudBackup(
-            password,
+      // If we still can't get the password, handle password not found
+      if (!fetchedPassword) {
+        handlePasswordNotFound && handlePasswordNotFound();
+        return;
+      }
+
+      dispatch(setIsWalletLoading(walletLoadingStates.BACKING_UP_WALLET));
+
+      // We have the password and we need to add it to an existing backup
+      logger.log('password fetched correctly');
+
+      let updatedBackupFile = null;
+
+      if (createBackupFileIfNeeded && fetchedPassword && !latestBackup) {
+        logger.log('backing up to icloud', wallets[walletId]);
+        updatedBackupFile = await backupWalletToCloud(
+          fetchedPassword,
+          wallets[walletId]
+        );
+      }
+
+      try {
+        if (latestBackup && !updatedBackupFile) {
+          updatedBackupFile = await addWalletToCloudBackup(
+            fetchedPassword,
             wallets[walletId],
             latestBackup
           );
         }
-        if (backupFile) {
-          logger.log('walletCloudBackup:: saving backup password');
-          await saveBackupPassword(password);
-          logger.log('walletCloudBackup:: saved');
+      } catch (e) {
+        onError && onError();
+        logger.sentry('error while trying to add wallet to icloud backup');
+        captureException(e);
+        setTimeout(() => {
+          Alert.alert('Error while trying to backup');
+        }, 500);
+      }
 
-          logger.log('walletCloudBackup:: backup completed!', backupFile);
+      try {
+        if (updatedBackupFile) {
+          logger.log('backup completed!');
           await dispatch(
-            setWalletBackedUp(walletId, walletBackupTypes.cloud, backupFile)
+            setWalletBackedUp(
+              walletId,
+              WalletBackupTypes.cloud,
+              updatedBackupFile
+            )
           );
-          logger.log('walletCloudBackup:: backup saved everywhere!');
+          logger.log('backup saved everywhere!');
           onSuccess && onSuccess();
-          goBack();
         } else {
-          onError();
+          onError && onError();
+          setTimeout(() => {
+            Alert.alert('Error while trying to backup');
+          }, 500);
         }
       } catch (e) {
-        onError(e);
+        logger.sentry('error while trying to save wallet backup state');
+        captureException(e);
```

Comment:
Should we have an alert here too?

Reply:
We might want to revisit redux/wallets.js setWalletBackedUp and improve our error handling there